data-loader/core/src/main/java/com/scalar/db/dataloader/core/tablemetadata/TableMetadataService.java

```diff
@@ -0,0 +1,75 @@
+package com.scalar.db.dataloader.core.tablemetadata;
+
+import com.scalar.db.api.DistributedStorageAdmin;
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.dataloader.core.util.TableMetadataUtil;
+import com.scalar.db.exception.storage.ExecutionException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Service for retrieving {@link TableMetadata} from ScalarDB. Provides methods to fetch metadata
+ * for individual tables or a collection of tables.
+ */
+@RequiredArgsConstructor
+public class TableMetadataService {
+
+  private static final String ERROR_MISSING_NAMESPACE_OR_TABLE =
+      "Missing namespace or table: %s, %s";
```

Comment:
Please move this to `CoreError`. Or we might already have the same message for it in `CoreError`. If so, pease use it.